src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -2196,6 +2224,25 @@ export class SavedObjectsRepository {
       );
     }
   }
+
+  /** Validate a migrated doc against the registered saved object type's schema. */
+  private validateObjectAttributes<T>(type: string, doc: SavedObjectSanitizedDoc<T>) {
+    const savedObjectType = this._registry.getType(type);
+    if (!savedObjectType?.schemas) {
```

Comment:
Shouldn't we log a warning if there isn't a validation schema declared? That way, we'll have some way of monitoring migration failures (or any other failures from malformed SO creation actions) over time and address the critical ones that cause issues.

Reply:
If we do that, we need to enable it in dev mode only, as it's a developer warning and not a production one.

It could be very verbose though, and I'm not sure it would replace the usual meta issue pinging all the type owners anyway?

Reply:
> It could be very verbose though, and I'm not sure it would replace the usual meta issue pinging all the type owners anyway?

Verbosity is what we want in debug mode but you're right, we don't know if it will help in the long run or be more "noise" than we can reasonably sift through.

Reply:
How are we going to track the SO without validation? How soon do we want to get other teams to cover SO with validation?

Reply:
Considering that with the latest updates to this PR we are now performing validation on the whole SO regardless of whether any validations have been registered for the attributes, IMO it is okay to do the usual pinging of type owners so that they can decide how to prioritize adding their own validations for the actual attributes. I think for some teams this will likely be a much larger concern than for others, so I don't think we necessarily need to put a timetable on it yet...